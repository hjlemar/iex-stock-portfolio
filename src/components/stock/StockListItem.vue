<template>
  <v-list-tile class="test">
    <v-list-tile-content @click="showQuote()">
      <v-list-tile-title>{{ stock }} 10.00</v-list-tile-title>
    </v-list-tile-content>
    <v-list-tile-action>
      <!-- <v-btn ripple round small @click.native.stop="showPopUp">Add</v-btn> -->
      <app-stock-dialog
        :cancel="cancel"
        :item="item"
      ></app-stock-dialog>
    </v-list-tile-action>
  </v-list-tile>
</template>

<script>
import { mapActions } from 'vuex';
import { GET_DETAILS } from '@/store/modules/details/types';
import StockDialog from '@/components/stock/StockDialog';

export default {
  props: ['stock'],
  components: {
    appStockDialog: StockDialog,
  },
  data() {
    return {
      dialog: false,
      item: {},
    };
  },
  methods: {
    ...mapActions({
      getQuote: GET_DETAILS,
    }),
    showQuote() {
      const stock = this.stock;
      this.getQuote(stock)
        .then(() => this.$router.push({ path: `/details/${stock}` }));
    },
    showPopUp() {
      this.item = { symbol: this.stock, shares: 0 };
      this.dialog = true;
    },
    cancel() {
      this.dialog = false;
    },
    save() {
      this.dialog = false;
    },
  },
};
</script>

<style>
  .test:hover {
    background-color: #a442f4;
    cursor: pointer;
  }
</style>

