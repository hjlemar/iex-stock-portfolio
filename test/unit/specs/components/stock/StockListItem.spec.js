import { shallow, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import StockListItem from '@/components/stock/StockListItem';
import Vuetify from 'vuetify';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(Vuetify);

describe('StockListItem', () => {
  let store;

  beforeEach(() => {
    store = new Vuex.Store({ });
  });

  it('should render component and match snapshot', () => {
    const wrapper = shallow(StockListItem, { store, localVue });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
