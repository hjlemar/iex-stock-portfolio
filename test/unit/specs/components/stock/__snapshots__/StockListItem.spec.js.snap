// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockListItem should render component and match snapshot 1`] = `undefined`;

exports[`StockListItem should render component and match snapshot with stock O 1`] = `
<div class="test">
    <div class="list__tile">
        <div class="list__tile__content">
            <div class="list__tile__title">O 10.00</div>
        </div>
        <div class="list__tile__action">
            <button type="button" class="btn btn--round btn--small">
                <div class="btn__content">Add</div>
            </button>
        </div>
    </div>
</div>
`;
